### PR TITLE
Update WorldEdit for Sponge to API 9 (1.18.2)

### PR DESCRIPTION
**Pull request by Lignium**
This is my attempt to port WorldEdit for Sponge to version 1.18.2 (API 9).

- [x] Fix syntax errors
- [x] Fix Gradle build
- [x] Fix `SpongeTransmogrifier`

### CODE REVIEW

File: worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/internal/SpongeWorldNativeAccess.java [hunk 1]
```diff
@@ -105,8 +105,8 @@ public boolean updateTileEntity(BlockPos position, com.sk89q.jnbt.CompoundTag ta
         if (tileEntity == null) {
             return false;
         }
-        tileEntity.setLevelAndPosition(getWorld(), position);
```
Comment (Lignium) [inline worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/internal/SpongeWorldNativeAccess.java:108]: What is the point of calling `setLevelAndPosition` here?

File: worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/SpongeWorld.java [hunk 2]
```diff
@@ -285,43 +286,43 @@ public boolean regenerate(Region region, Extent extent, RegenOptions options) {
...
+                return TreeFeatures.SWAMP_OAK.value();
             case ACACIA:
-                return Features.ACACIA;
+                return TreeFeatures.ACACIA.value();
             case DARK_OAK:
-                return Features.DARK_OAK;
+                return TreeFeatures.DARK_OAK.value();
             case TALL_BIRCH:
-                return Features.BIRCH_TALL;
+                return TreeFeatures.SUPER_BIRCH_BEES.value(); // TODO Bees???
```
Comment (Lignium) [inline worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/SpongeWorld.java:315]: I cannot find the exact equivalent.

File: worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/internal/SpongeWorldNativeAccess.java (hunk 1 above)
Comment (me4502) [reply worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/internal/SpongeWorldNativeAccess.java:108]: IIRC that data was missing from the BlockEntity instance

Comment (Lignium) [reply worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/internal/SpongeWorldNativeAccess.java:108]: > IIRC that data was missing from the BlockEntity instance

In this game version, position cannot be set, only level

Comment (me4502) [reply worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/internal/SpongeWorldNativeAccess.java:108]: Is the position data available in the BlockEntity though? In the old version, that data was missing and had to be added

**Review by me4502 — COMMENTED** [review]
This PR looks mostly good outside of the pending question; however I'm unsure how we're wanting to handle merging Sponge API 9. Is this a version that's considered "released" or near releasing? As the version/7.2.x branch is our "releases" branch, anything that goes in here should be considered something we're happy to have as a newly cut release

**Pull request merged**